### ISSUE TITLE
Compression doesn't seems to be consistent

### ISSUE THREAD

**Issue by kdybicz** [jaredwray/keyv#498]
I was considering updating `keyv` library to the latest version and switching the custom compression wrapper to one of the new packages. Unfortunately I found couple of potential issues I would like to discuss.

1. Looks like:
```
compress?: Record<string, unknown> | undefined;
```
from [Keyv.Options interface](https://github.com/jaredwray/keyv/blob/main/packages/keyv/src/index.d.ts#L88) are no longer used in code.

2. New `compression` parameter used in the [Keyv implementation](https://github.com/jaredwray/keyv/blob/main/packages/keyv/src/index.js#L53-L58) doesn't seem to have a clear interface that could be used to define the `KeyvBrotli`, `KeyvGzip` or any other future implementations.

3. The approach of using `opts` object created inside `KeyvBrotli` and `KeyvGzip` instead of methods defined by common interface seems to be a dodgy approach. It doesn't make it easier to work on future integrations.

4. Both `KeyvBrotli` and `KeyvGzip` seems to be defining only two methods in [Options interface](https://github.com/jaredwray/keyv/blob/main/packages/compress-brotli/src/index.d.ts#L14-L15) that when passed to the constructor are [immediately overwritten](https://github.com/jaredwray/keyv/blob/main/packages/compress-brotli/src/index.js#L12-L15). Additionally compress and decompress methods are being defined in those two classes, which makes it even harder to understand how the code should behave.

5. I'm not sure if either of those `extends EventEmitter implements Store<Value>` is really needed in `KeyvBrotli` and `KeyvGzip`.

6. There is not TypeScript compatible way to provide compression-method-related parameters down to the compressors.

I know that's a lot, but I think current approach is bugged and I would be happy to jump on a call to talk all of those points through.

**Comment by jaredwray** [comment]
@kdybicz - happy to have a discussion around this as this was our first attempt to get this going and also keep compatibility with all the storage adapters on the backend. Here are some initial thoughts:

```
Looks like:
compress?: Record<string, unknown> | undefined;
from [Keyv.Options interface](https://github.com/jaredwray/keyv/blob/main/packages/keyv/src/index.d.ts#L88) are no longer used in code.
```
>  Correct, our plan will be to remove this on the next release. I will get a pull request going on it. 

```
New compression parameter used in the [Keyv implementation](https://github.com/jaredwray/keyv/blob/main/packages/keyv/src/index.js#L53-L58) doesn't seem to have a clear interface that could be used to define the KeyvBrotli, KeyvGzip or any other future implementations.
```
> Good point, Right now we have attempted to make this more flexible but then harden it over time. Our current thinking is to define a pretty standard interface for people to build off of. 

```
Both KeyvBrotli and KeyvGzip seems to be defining only two methods in [Options interface](https://github.com/jaredwray/keyv/blob/main/packages/compress-brotli/src/index.d.ts#L14-L15) that when passed to the constructor are [immediately overwritten](https://github.com/jaredwray/keyv/blob/main/packages/compress-brotli/src/index.js#L12-L15). Additionally compress and decompress methods are being defined in those two classes, which makes it even harder to understand how the code should behave.
```

> Agree, What we will look at is to adjust this to not overwrite as you mentioned in #3 and also will look to use the compression directly if it is implemented. 

```
I'm not sure if either of those extends EventEmitter implements Store<Value> is really needed in KeyvBrotli and KeyvGzip.
```
> Will validate and see if we can change this. 

@alphmth - will get some time with you to discuss the changes.

**Comment by jaredwray** [comment]
@kdybicz happy to discuss via a zoom or call if that helps. Shoot me an email and we can setup some time.

**Comment by kdybicz** [comment]
@jaredwray I proposed a call just in case I did a bad job at explaining the points. If all is clear, then I leave that up to you. Also, feel free to ping me if you would need help ie. with testing those changes.

**Comment by jaredwray** [comment]
@kdybicz - will be working on some clean up for this in a pull request and will tag you in on it.

**Comment by jaredwray** [comment]
@kdybicz - https://github.com/jaredwray/keyv/pull/511

**Comment by jaredwray** [comment]
@kdybicz - one thing with this architecture is that it will overwrite the serialize and deserialize. We are looking at if this makes sense and if we want to make it more optional to do that which would increase some complexity in the main Keyv code base.

**Comment by jaredwray** [comment]
@kdybicz - will be closing this as the v2 architecture is now in main and will be released soon. We decided to keep the overwrite of `serialize` and `deserialize` functions in the compress function to simplify the current architecture in Keyv. With that said we will evaluate over time and might move to just needing `compress` and `decompress`.

In addition we have added an interface for this to standardize and have updated more of the code to work better and validated it.
